app/experimenter/experiments/tests/api/v5/test_queries.py

```diff
@@ -1005,6 +1005,98 @@ def test_targeting_config_slug_for_valid_targeting_config_returns_name(self):
+            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+        content = json.loads(response.content)
+        experiment_data = content["data"]["experimentBySlug"]
+        print(experiment_data)
+        self.assertEqual(
+            experiment_data["targetingConfig"],
+            [
+                {
+                    "label": NimbusExperiment.TargetingConfig.TARGETING_FIRST_RUN.label,
+                    "value": NimbusExperiment.TargetingConfig.TARGETING_FIRST_RUN.value,
+                    "applicationValues": list(
+                        NimbusExperiment.TARGETING_CONFIGS[
+                            NimbusExperiment.TargetingConfig.TARGETING_FIRST_RUN.value
+                        ].application_choice_names
+                    ),
+                    "description": NimbusExperiment.TARGETING_CONFIGS[
+                        NimbusExperiment.TargetingConfig.TARGETING_FIRST_RUN.value
+                    ].description,
+                }
+            ],
+        )
+
+    def test_targeting_configs_with_empty_targeting_config_slug(self):
+        user_email = "user@example.com"
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.CREATED,
+            application=NimbusExperiment.Application.DESKTOP,
+            targeting_config_slug="",
+        )
+
+        response = self.query(
+            """
+            query experimentBySlug($slug: String!) {
+                experimentBySlug(slug: $slug) {
+                    targetingConfig {
+                        label
+                        value
+                        description
+                        applicationValues
+                    }
+                }
+            }
+            """,
+            variables={"slug": experiment.slug},
+            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+        content = json.loads(response.content)
+        experiment_data = content["data"]["experimentBySlug"]
+        print(experiment_data)
+        self.assertEqual(
+            experiment_data["targetingConfig"],
+            [
+                {
+                    "label": NimbusExperiment.TargetingConfig.NO_TARGETING.label,
+                    "value": NimbusExperiment.TargetingConfig.NO_TARGETING.value,
+                    "applicationValues": list(
+                        NimbusExperiment.TARGETING_CONFIGS[
+                            NimbusExperiment.TargetingConfig.NO_TARGETING.value
+                        ].application_choice_names
+                    ),
+                    "description": NimbusExperiment.TARGETING_CONFIGS[
+                        NimbusExperiment.TargetingConfig.NO_TARGETING.value
+                    ].description,
+                }
+            ],
+        )
```

Comment:
This test is redundant since it's just testing a particular value rather than a special case, you can take this out I think.

Reply:
I added this to show if we don't have slug, in that case we get the default t`arget_config  as NO_TARGETING`, should we remove it?

Reply:
Yeah, if the field defaulted to `None` then we'd be concerned about handling that case correctly since the dict key lookup would fail, but since the field defaults to a valid choice and isn't nullable, we never need to worry about that case so we can get away with the single test 👍

Reply:
got it makes sense :)